doc/guide/sample_ml_app.hpp

```diff
@@ -0,0 +1,194 @@
+
+@section sample_create_project Creating the VS project
+
+- Open Visual Studio and create a new project (Windows Console Application)
+- For this sample, the project is named “sample-ml-app”
+
+@section sample_project_config Project Configuration
+
+There are different ways in which you can configure your project to link with dependencies. This configuration
+is for x64 Debug Mode. If you need Release Mode, please change the paths accordingly (assuming you have built
+mlpack and dependencies in Release Mode).
+
+- Right click on the project and select Properties, select the x64 Debug profile
+- Under C/C++ > General > Additional Include Directories add:
+@code
+ - C:\boost\boost_1_66_0
+ - C:\mlpack\armadillo-8.500.1\include
+ - C:\mlpack\mlpack-3.0.2\build\include
+@endcode
+- Under Linker > Input > Additional Dependencies add:
+@code
+ - C:\mlpack\mlpack-3.0.2\build\Debug\mlpack.lib
+ - C:\boost\boost_1_66_0\lib64-msvc-14.1\libboost_serialization-vc141-mt-gd-x64-1_66.lib
+ - C:\boost\boost_1_66_0\lib64-msvc-14.1\libboost_program_options-vc141-mt-gd-x64-1_66.lib
+@endcode
+- Under Build Events > Post-Build Event > Command Line add:
+@code
+ - xcopy /y "C:\mlpack\mlpack-3.0.2\build\Debug\mlpack.dll" $(OutDir)
+ - xcopy /y "C:\mlpack\mlpack-3.0.2\packages\OpenBLAS.0.2.14.1\lib\native\bin\x64\*.dll" $(OutDir)
+@endcode
+
+@note Recent versions of Visual Studio set "Conformance Mode" enabled by default. This causes some issues with
+the armadillo library. If you encounter this issue, disable "Conformance Mode" under C/C++ > Language.
+
+@section sample_app_goal The app goal
+
+This app aims to exercise an end-to-end machine learning workflow. We will cover:
+
+- Loading and preparing a dataset
+- Training (using Random Forest as example)
+- Computing the training accuracy
+- Cross-Validation using K-Fold
+- Metrics gathering (accuracy, precision, recall, F1)
+- Saving the trained model to disk
+- Loading the model
+- Classifying a new sample
+
+@section sample_headers_namespaces Headers and namespaces
+
+For this app, we will need to include the following headers (i.e. add into stdafx.h):
+
+@code
+#include "mlpack/core.hpp"
+#include "mlpack/methods/random_forest/random_forest.hpp"
+#include "mlpack/methods/decision_tree/random_dimension_select.hpp"
+#include "mlpack/core/cv/k_fold_cv.hpp"
+#include "mlpack/core/cv/metrics/accuracy.hpp"
+#include "mlpack/core/cv/metrics/precision.hpp"
+#include "mlpack/core/cv/metrics/recall.hpp"
+#include "mlpack/core/cv/metrics/F1.hpp"
+@endcode
+
+Also, we will use the following namespaces:
+
+@code
+using namespace arma;
+using namespace mlpack;
+using namespace mlpack::tree;
+using namespace mlpack::cv;
+@endcode
+
+@section sample_load_dataset Loading the dataset
+
+First step is about loading the dataset. Here we load a CSV dataset 
+(i.e.: mlpack/tests/data/german.csv), assuming the labels don't require normalization.
+
+@code
+bool loaded = mlpack::data::Load("data/german.csv", dataset);
```

Comment:
Should we define ``dataset, labels`` first? I guess people might copy each line, and end up with some errors.

Reply:
Yep, added the missing dataset definition.